src/plugins/vis_type_pie/public/editor/components/pie.tsx

```diff
@@ -0,0 +1,222 @@
+
+export interface PieOptionsProps extends VisEditorOptionsProps<PieVisParams>, PieTypeProps {}
+
+const PieOptions = (props: PieOptionsProps) => {
+  const { stateParams, setValue, aggs } = props;
+  const setLabels = <T extends keyof PieVisParams['labels']>(
+    paramName: T,
+    value: PieVisParams['labels'][T]
+  ) => setValue('labels', { ...stateParams.labels, [paramName]: value });
+  const legendUiStateValue = props.uiState?.get('vis.legendOpen');
+  const [palettesRegistry, setPalettesRegistry] = useState<PaletteRegistry | undefined>(undefined);
+  const [legendVisibility, setLegendVisibility] = useState<boolean>(() => {
+    const bwcLegendStateDefault = stateParams.addLegend == null ? false : stateParams.addLegend;
+    return props.uiState?.get('vis.legendOpen', bwcLegendStateDefault) as boolean;
+  });
+  const hasSplitChart = aggs?.aggs?.find((agg) => agg.schema === 'split' && agg.enabled);
+
+  useEffect(() => {
+    setLegendVisibility(legendUiStateValue);
+  }, [legendUiStateValue]);
+
+  useEffect(() => {
+    const fetchPalettes = async () => {
+      const palettes = await props.palettes?.getPalettes();
+      setPalettesRegistry(palettes);
+    };
+    fetchPalettes();
+  }, [props.palettes]);
+
+  return (
+    <>
+      <EuiPanel paddingSize="s">
+        <EuiTitle size="xs">
+          <h3>
+            <FormattedMessage
+              id="visTypePie.editors.pie.pieSettingsTitle"
+              defaultMessage="Pie settings"
+            />
+          </h3>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <SwitchOption
+          label={i18n.translate('visTypePie.editors.pie.donutLabel', {
+            defaultMessage: 'Donut',
+          })}
+          paramName="isDonut"
+          value={stateParams.isDonut}
+          setValue={setValue}
+        />
+        <BasicOptions {...props} legendPositions={getLegendPositions} />
+        {props.showElasticChartsOptions && (
+          <>
+            <SwitchOption
+              label={i18n.translate('visTypePie.editors.pie.distinctColorsLabel', {
+                defaultMessage: 'Distinct colors per slice',
+              })}
+              paramName="distinctColors"
+              value={stateParams.distinctColors}
+              setValue={setValue}
+              data-test-subj="visTypePiedistinctColorsSwitch"
+            />
+            <SwitchOption
+              label={i18n.translate('visTypePie.editors.pie.addLegendLabel', {
+                defaultMessage: 'Show legend',
+              })}
+              paramName="addLegend"
+              value={legendVisibility}
+              setValue={(paramName, value) => {
+                setLegendVisibility(value);
+                setValue(paramName, value);
+              }}
+              data-test-subj="visTypePieAddLegendSwitch"
+            />
+            <SwitchOption
+              label={i18n.translate('visTypePie.editors.pie.nestedLegendLabel', {
+                defaultMessage: 'Nested legend',
```

Comment:
Can this toggle only get enabled when the legend is visible?

Reply:
Done!